src/components/rbac/permission-categories.tsx

```diff
@@ -0,0 +1,145 @@
+import * as React from 'react';
+import { t } from '@lingui/macro';
+import { AppContext } from 'src/loaders/app-context';
+import { RoleType, GroupRoleType, PermissionType } from 'src/api';
+import { PermissionChipSelector } from 'src/components';
+import { Constants } from 'src/constants';
+import { Flex, FlexItem } from '@patternfly/react-core';
+
+interface IState {
+  groups: PermissionType[];
+}
+
+interface IProps {
+  showEmpty: boolean;
+  showCustom: boolean;
+  role: RoleType | GroupRoleType;
+  showUserMgmt?: boolean;
+}
+
+export class PermissionCategories extends React.Component<IProps, IState> {
+  static contextType = AppContext;
+  constructor(props) {
+    super(props);
+    this.state = {
+      groups: [],
+    };
+  }
+
+  componentDidMount() {
+    const { model_permissions } = this.context.user;
+    this.setState({ groups: this.formatPermissions(model_permissions) });
+  }
+
+  render() {
+    const { groups } = this.state;
+    const { role, showEmpty, showCustom, showUserMgmt } = this.props;
+    const { model_permissions } = this.context.user;
+
+    if (!showUserMgmt) {
+      Constants.USER_GROUP_MGMT_PERMISSIONS.forEach((perm) => {
+        if (perm in model_permissions) {
+          delete model_permissions[perm];
+        }
+      });
+    }
```

Comment:
oh this is problematic because we're mutating state in context by doing it this way... after this deletes a permission, permission checks for that permission will start failing because it's no longer a known permission

.. so we need to copy the object first, before changing it..

(and delete doesn't mind if the permission doesn't exist)

```suggestion
    const model_permissions = { ...this.context.user.model_permissions };

    if (!showUserMgmt) {
      Constants.USER_GROUP_MGMT_PERMISSIONS.forEach(
        (perm) => delete model_permissions[perm],
      );
    }
```